### PR TITLE
[Root]: Added default color for factions + added two new types.

### PR DESCRIPTION
**Pull request by Kuroosh**
This will add a default color for factions + the new types `medic` & `mechanic`.
The color PR will be needed for the color palette in the /fopen window ( in the next FR ) ( for faction chat color, gangwar, etc etc. )

### CODE REVIEW

**Comment by Progefix** [comment]
medic is a state faction. i guess stuykt dont wont specific factions stuff like lspd lsmd etc.

**Comment by Kuroosh** [comment]
> medic is a state faction. i guess stuykt dont wont specific factions stuff like lspd lsmd etc.

Well... yes & no.
medics should not be able to arrest people.
lspd should not be able to revive other people xd

**Comment by Kuroosh** [comment]
checking it in a non-dynamic way ( by id ) isn't very nice.. 
so if a faction has a type ( like state ( LSPD, FBI, Army ) factions ) 
its easier to interact with that.

**Comment by Stuyk** [comment]
I think this could be implemented in a way that would allow the expansion of faction types without touching it directly or adding opinionated types to it like; oh we need EMT cause that's what our gamemode needs.

ie. Change `type` to
```
type: "x" | "y" | "z";
```

Can probably pass a template type to the Faction interface itself.

Below is just mock code of what I'm thinking.
```ts
interface <T>Faction {
    ...
    type: "x" | "y" | "z" & T?
}
```

File: src/core/plugins/core-factions/shared/interfaces.ts [hunk 1]
```diff
@@ -235,6 +235,14 @@ export interface FactionCore {
...
      */
     type: string;
+
+    /**
+     * Faction Color
+     *
+     * @type {Color}
+     * @memberof Faction
+     */
+    color: Color;
```
Comment (Stuyk) [inline src/core/plugins/core-factions/shared/interfaces.ts:245]: Use `alt.RGBA`

File: src/core/plugins/core-factions/shared/interfaces.ts [hunk 2]
```diff
@@ -235,6 +235,14 @@ export interface FactionCore {
      * @memberof Faction
      */
     type: string;
```
Comment (Stuyk) [inline src/core/plugins/core-factions/shared/interfaces.ts:237]: Use a Template Type that can be passed to the Faction interface with a default of the `FactionTypes`.

**Review by Stuyk — CHANGES_REQUESTED** [review]

File: src/core/plugins/core-factions/shared/interfaces.ts (hunk 1 above)
Comment (Kuroosh) [reply src/core/plugins/core-factions/shared/interfaces.ts:245]: Sure, new PR will coming soon :)

**Pull request closed without merging**